test.sh

```diff
@@ -351,7 +353,14 @@ function test_rustc() {
     git checkout tests/ui/type-alias-impl-trait/auxiliary/cross_crate_ice2.rs
     git checkout tests/ui/macros/rfc-2011-nicer-assert-messages/auxiliary/common.rs
 
-    RUSTC_ARGS="$TEST_FLAGS -Csymbol-mangling-version=v0 -Zcodegen-backend="$(pwd)"/../target/"$CHANNEL"/librustc_codegen_gcc."$dylib_ext" --sysroot "$(pwd)"/../build_sysroot/sysroot"
+    # We need to overwrite the sysroot in the tests, now.
+    # TODO(antoyo): find a faster way to do this.
+    # FIXME: this makes the stderr different since it changes the line numbers.
+    for file in $(find tests/ui -type f -name '*.rs'); do
+        sed -ie "1i // compile-flags: --sysroot "$(pwd)"/../build_sysroot/sysroot\n" $file
+    done
+
+    RUSTC_ARGS="$TEST_FLAGS -Csymbol-mangling-version=v0 -Zcodegen-backend="$(pwd)"/../target/"$CHANNEL"/librustc_codegen_gcc."$dylib_ext""
```

Comment:
Add back the `--sysroot` option when https://github.com/rust-lang/rust/pull/112435 is merged.